### ISSUE TITLE
Tests - Unit Tests

### ISSUE THREAD

**Issue by marcelomogrovejo** [marcelomogrovejo/LoginDemo.SwiftUI#34]
Figure out how to do tests on SwiftUI... maybe UI Tests ?

**Comment by marcelomogrovejo** [comment]
Add accessibilityIdentifier() ?

**Comment by marcelomogrovejo** [comment]
Because the topics are a little big, it is a good idea to split the tasks in Unit Tests and UI Tests.

**Comment by marcelomogrovejo** [comment]
Add Unit tests and a test plan that includes code coverage.

**Comment by marcelomogrovejo** [comment]
Lecture: https://www.kodeco.com/30227776-swiftui-testing-with-viewinspector-for-ios
Implementing https://github.com/nalexn/ViewInspector to test UI.

**Comment by marcelomogrovejo** [comment]
Implementing https://github.com/nalexn/ViewInspector to test UI.

**Comment by marcelomogrovejo** [comment]
As it happened on the implementation of ViewInspector demo app, the are several purple warnings related to StateObject, view model:

Accessing StateObject<SignInViewModel>'s object without being installed on a View. This will create a new instance each time.

The test are working but it is needed to figure out how to solve this kind of warnings.

**Comment by marcelomogrovejo** [comment]
I have decided to stop here the implementation of ViewInspector testing because it looks like the library is in development state yet.
Image component doesn't have accessibilityIdentifier and accessibilityLabel components.
button.tap() doesn't work on toggle.
etc.

Versions:
Xcode 16.3
ViewInspector 0.10.1
Swift 6.1 ($ xcrun swift -version)
So, for now, I decided to go forward and implement some UI tests instead.

**The branch and the task will remind open.**
